pkgs/applications/video/dvdstyler/default.nix

```diff
@@ -1,84 +1,106 @@
-{ lib, stdenv, fetchurl, pkg-config
-, flex, bison, gettext
-, xineUI, wxSVG
+{ lib
+, stdenv
+, fetchurl
+, bison
+, cdrtools
+, docbook5
+, dvdauthor
+, dvdplusrwtools
+, flex
 , fontconfig
-, xmlto, docbook5, zip
-, cdrtools, dvdauthor, dvdplusrwtools
+, gettext
+, makeWrapper
+, pkg-config
+, wxSVG
+, xine-ui
+, xmlto
+, zip
+
 , dvdisasterSupport ? true, dvdisaster ? null
 , thumbnailSupport ? true, libgnomeui ? null
 , udevSupport ? true, udev ? null
 , dbusSupport ? true, dbus ? null
-, makeWrapper }:
-
-with lib;
-stdenv.mkDerivation rec {
+}:
 
+let
+  inherit (lib) optionals makeBinPath;
+in stdenv.mkDerivation rec {
   pname = "dvdstyler";
-  srcName = "DVDStyler-${version}";
   version = "3.1.2";
 
   src = fetchurl {
-    url = "mirror://sourceforge/project/dvdstyler/dvdstyler/${version}/${srcName}.tar.bz2";
+    url = "mirror://sourceforge/project/dvdstyler/dvdstyler/${version}/DVDStyler-${version}.tar.bz2";
     sha256 = "03lsblqficcadlzkbyk8agh5rqcfz6y6dqvy9y866wqng3163zq4";
   };
 
-  nativeBuildInputs =
-  [ pkg-config ];
-
-  packagesToBinPath =
-  [ cdrtools dvdauthor dvdplusrwtools ];
-
-  buildInputs =
-  [ flex bison gettext xineUI
-    wxSVG fontconfig xmlto
-    docbook5 zip makeWrapper ]
-  ++ packagesToBinPath
+  nativeBuildInputs = [
+    pkg-config
+  ];
+  buildInputs = [
+    bison
+    cdrtools
+    docbook5
+    dvdauthor
+    dvdplusrwtools
+    flex
+    fontconfig
+    gettext
+    makeWrapper
+    wxSVG
+    xine-ui
+    xmlto
+    zip
+ ]
   ++ optionals dvdisasterSupport [ dvdisaster ]
   ++ optionals udevSupport [ udev ]
   ++ optionals dbusSupport [ dbus ]
   ++ optionals thumbnailSupport [ libgnomeui ];
 
-  binPath = makeBinPath packagesToBinPath;
 
-  postInstall = ''
-    wrapProgram $out/bin/dvdstyler \
-      --prefix PATH ":" "${binPath}"
-  '';
+  postInstall = let
+    binPath = makeBinPath [
+      cdrtools
+      dvdauthor
+      dvdplusrwtools
+    ]; in ''
+       wrapProgram $out/bin/dvdstyler --prefix PATH ":" "${binPath}"
```

Comment:
```suggestion
    binPath = ''
       wrapProgram $out/bin/dvdstyler --prefix PATH ":" "${makeBinPath [cdrtools dvdauthor dvdplusrwtools ];}"
```

Reply:
I would say is preferential, and defer to the maintainer of the package. The former would be easier to edit in the future.